featuretools/tests/synthesis/test_koalas_dfs.py

```diff
@@ -0,0 +1,379 @@
+
+    ks_fm, _ = ft.dfs(entityset=ks_es,
+                      target_entity="data",
+                      trans_primitives=primitives_list,
+                      cutoff_time=cutoff_times)
+
+    pd_es = ft.EntitySet(id="pd_es")
+    pd_es.entity_from_dataframe(entity_id="data",
+                                dataframe=df,
+                                index="id",
+                                time_index="dates",
+                                variable_types={"strings": ft.variable_types.NaturalLanguage})
+
+    fm, _ = ft.dfs(entityset=pd_es,
+                   target_entity="data",
+                   trans_primitives=primitives_list,
+                   cutoff_time=cutoff_times)
+    # Because row ordering with koalas is not guaranteed, `we need to sort on two columns to make sure that values
+    # for instance id 0 are compared correctly. Also, make sure the boolean column has the same dtype.
+    fm = fm.sort_values(['id', 'labels'])
+    ks_fm = ks_fm.to_pandas().set_index('id').sort_values(['id', 'labels'])
+    ks_fm['IS_WEEKEND(dates)'] = ks_fm['IS_WEEKEND(dates)'].astype(fm['IS_WEEKEND(dates)'].dtype)
+    pd.testing.assert_frame_equal(fm, ks_fm)
+
+
+@pytest.mark.skipif('not ks')
+def test_single_table_ks_entityset_dates_not_sorted():
+    ks_es = EntitySet(id="ks_es")
+    df = pd.DataFrame({"id": [0, 1, 2, 3],
+                       "values": [1, 12, -34, 27],
+                       "dates": [pd.to_datetime('2019-01-10'),
+                                 pd.to_datetime('2019-02-03'),
+                                 pd.to_datetime('2019-01-01'),
+                                 pd.to_datetime('2017-08-25')]})
+
+    primitives_list = ['absolute', 'is_weekend', 'year', 'day']
+    values_dd = ks.from_pandas(df)
+    vtypes = {
+        "id": ft.variable_types.Id,
+        "values": ft.variable_types.Numeric,
+        "dates": ft.variable_types.Datetime,
+    }
+    ks_es.entity_from_dataframe(entity_id="data",
+                                dataframe=values_dd,
+                                index="id",
+                                time_index="dates",
+                                variable_types=vtypes)
+
+    ks_fm, _ = ft.dfs(entityset=ks_es,
+                      target_entity="data",
+                      trans_primitives=primitives_list,
+                      max_depth=1)
+
+    pd_es = ft.EntitySet(id="pd_es")
+    pd_es.entity_from_dataframe(entity_id="data",
+                                dataframe=df,
+                                index="id",
+                                time_index="dates")
+
+    fm, _ = ft.dfs(entityset=pd_es,
+                   target_entity="data",
+                   trans_primitives=primitives_list,
+                   max_depth=1)
+
+    pd.testing.assert_frame_equal(fm, ks_fm.to_pandas().set_index('id').loc[fm.index])
+
+
+@pytest.mark.skipif('not ks')
+def test_secondary_time_index():
```

Comment:
same as with dask test, let's make this `test_ks_entityset_secondary_time_index`

Reply:
renamed